db/crud.go

```diff
@@ -869,6 +869,46 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 	return nil
 }
 
+// updateMutateInSpec updates the mutate in options with the appropriate mutate in spec needed based off the outcome in updateHLV
```

Comment:
nit: clarify that we're not modifying something passed in - but returning a slice to be passed into the MutateIn options

```suggestion
// updateMutateInSpec returns the mutate in spec needed for the document update based off the outcome in updateHLV
```